projects/wavelet_dataAggregation/checkDataType.py

```diff
@@ -0,0 +1,188 @@
+      row = rows[i].split(',')
+      row[0] = str(datetime.strptime(row[0], '%Y/%m/%d %H:%M:%S'))
+      newRows.append(row)
+
+    fileWriter = csv.writer(open(filePath, 'wb'))
+    fileWriter.writerows(newRows)
+
+  return fileNameList, filePathList, dataTypeList
+
+
+def estimateMedianAbsoluteDeviation(timestamp):
+  """
+  Estimate median sampling interval and the median absolute deviation (MAD)
+  of sampling interval
+  @param timestamp: numpy array of datetime that stores timestamp
+  @return medianSamplingInterval: median sampling interval
+          medianAbsoluteDev: median absolute deviation of sampling interval
+  """
+  samplingIntervals = np.diff(timestamp)
+  samplingIntervals = samplingIntervals.astype('float32')
+  samplingIntervals = samplingIntervals[samplingIntervals > 0]
+  medianSamplingInterval = np.median(samplingIntervals)
+  medianAbsoluteDev = np.median(np.abs(samplingIntervals - medianSamplingInterval))
+  return medianSamplingInterval, medianAbsoluteDev
+  # meanSamplingInterval = np.mean(samplingIntervals)
+  # samplingIntervalsStd = np.std(samplingIntervals)
+  # return meanSamplingInterval, samplingIntervalsStd
+
+
+def checkDataType(dataFilePath):
+  """
+  Return the data type (transactional or non-transactional)
+  The data type is determined via an data type indicator, defined as the
+  ratio between median absolute deviation and median of the sampling interval.
+
+  @param dataFilePath: Name of a csv file, the file must have two columns
+  with header "timestamp", and "value"
+  @return dataType: a string with value "transactional" or "non-transactional"
+  """
+  (timestamp, sig) = readCSVfiles(dataFilePath)
+  medianSamplingInterval, medianAbsoluteDev = estimateMedianAbsoluteDeviation(timestamp)
+
+  dataTypeIndicator = medianAbsoluteDev/medianSamplingInterval
+  if dataTypeIndicator > 0.2:
+    dataType = "transactional"
+  else:
+    dataType = "non-transactional"
+
+  return dataType, dataTypeIndicator
+
+
+def checkAlgorithmPerformance(dataTypeList, predictedDataTypeList, dataTypeIndicatorList):
+  """
+  Check the performance of the data type detection algorithm
+  :param dataTypeList: List of data type
+  :param predictedDataTypeList: List of predicted data type
+  :param dataTypeIndicator: List of the datatype indicator
+  """
+
+  numCorrect = 0
+  for i in xrange(len(dataTypeList)):
+    if dataTypeList[i] == predictedDataTypeList[i]:
+      numCorrect += 1
+
+  correctRate = float(numCorrect)/len(dataTypeList)
+  print "correct rate: ", str(correctRate)
+
+
+if __name__ == "__main__":
+
+  print "Test algorithm on river-view data streams ..."
+  fileNameList, filePathList, dataTypeList = downloadExampleData()
+  dataTypeIndicatorList = []
+  predictedDataTypeList = []
+  for i in xrange(len(filePathList)):
+    filePath = filePathList[i]
+    predictedDataType, dataTypeIndicator = checkDataType(filePath)
+    print " File: ", fileNameList[i], " DataType: ", dataTypeList[i], \
```

Comment:
Use brackets to wrap lines instead of `\`. Here:
```
    print (" File: ", fileNameList[i], " DataType: ", dataTypeList[i], 
          " predicted Data Type: ", predictedDataType)
```